testing nil error

The following program `sample.go` triggers a panic:

```go
package main

func test() error { return nil }

func main() {
        if err := test(); nil == err {
                println("err is nil")
        }
}
```

Expected result:
```console
$ go run ./sample.go
err is nil
```

Got:
```console
$ yaegi ./sample.go
/home/marc/tmp/sample.go:6:20: CFG post-order panic: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc00038e000, 0x40)
        /home/marc/sdk/go1.14.1/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc00009de10)
        /home/marc/dev/yaegi/interp/interp.go:320 +0xbc
panic(0xd71760, 0xc00000e760)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:967 +0x166
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2.1(0xc00038b800)
        /home/marc/dev/yaegi/interp/cfg.go:387 +0xb3
panic(0xcde940, 0x1542e10)
        /home/marc/sdk/go1.14.1/src/runtime/panic.go:973 +0x3e3
github.com/containous/yaegi/interp.isInt(...)
        /home/marc/dev/yaegi/interp/type.go:1207
github.com/containous/yaegi/interp.isNumber(0x0, 0x0, 0xcea7c0)
        /home/marc/dev/yaegi/interp/type.go:1245 +0x26
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2(0xc00038b800)
        /home/marc/dev/yaegi/interp/cfg.go:608 +0x9257
github.com/containous/yaegi/interp.(*node).Walk(0xc00038b800, 0xc00009dc90, 0xc00009dcb8)
        /home/marc/dev/yaegi/interp/interp.go:191 +0xaf
github.com/containous/yaegi/interp.(*node).Walk(0xc00038b300, 0xc00009dc90, 0xc00009dcb8)
        /home/marc/dev/yaegi/interp/interp.go:188 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc00038b200, 0xc00009dc90, 0xc00009dcb8)
        /home/marc/dev/yaegi/interp/interp.go:188 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc00038ad00, 0xc00009dc90, 0xc00009dcb8)
        /home/marc/dev/yaegi/interp/interp.go:188 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc00038a000, 0xc00009dc90, 0xc00009dcb8)
        /home/marc/dev/yaegi/interp/interp.go:188 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc000001800, 0xc00038a000, 0x7ffe28e3464a, 0x17, 0xc000034178, 0x4, 0x7ffe28e3464a, 0x17, 0x0)
        /home/marc/dev/yaegi/interp/cfg.go:56 +0x16d
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000001800, 0xc000386000, 0x7c, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/marc/dev/yaegi/interp/interp.go:343 +0x1ea
main.main()
        /home/marc/dev/yaegi/cmd/yaegi/yaegi.go:94 +0x548
```

Thanks to @nrwiersma for the report.

--- a/_test/nil2.go
+++ b/_test/nil2.go
@@ -1,0 +1,12 @@
+package main
+
+func test() error { return nil }
+
+func main() {
+	if err := test(); nil == err {
+		println("err is nil")
+	}
+}
+
+// Output:
+// err is nil

--- a/interp/type.go
+++ b/interp/type.go
@@ -1204,6 +1204,9 @@ func isStruct(t *itype) bool {
 func isBool(t *itype) bool { return t.TypeOf().Kind() == reflect.Bool }
 
 func isInt(t reflect.Type) bool {
+	if t == nil {
+		return false
+	}
 	switch t.Kind() {
 	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Uintptr:
 		return true
@@ -1212,6 +1215,9 @@ func isInt(t reflect.Type) bool {
 }
 
 func isUint(t reflect.Type) bool {
+	if t == nil {
+		return false
+	}
 	switch t.Kind() {
 	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Uintptr:
 		return true
@@ -1220,6 +1226,9 @@ func isUint(t reflect.Type) bool {
 }
 
 func isComplex(t reflect.Type) bool {
+	if t == nil {
+		return false
+	}
 	switch t.Kind() {
 	case reflect.Complex64, reflect.Complex128:
 		return true
@@ -1228,6 +1237,9 @@ func isComplex(t reflect.Type) bool {
 }
 
 func isFloat(t reflect.Type) bool {
+	if t == nil {
+		return false
+	}
 	switch t.Kind() {
 	case reflect.Float32, reflect.Float64:
 		return true
@@ -1236,11 +1248,14 @@ func isFloat(t reflect.Type) bool {
 }
 
 func isByteArray(t reflect.Type) bool {
+	if t == nil {
+		return false
+	}
 	k := t.Kind()
 	return (k == reflect.Array || k == reflect.Slice) && t.Elem().Kind() == reflect.Uint8
 }
 
-func isFloat32(t reflect.Type) bool { return t.Kind() == reflect.Float32 }
-func isFloat64(t reflect.Type) bool { return t.Kind() == reflect.Float64 }
+func isFloat32(t reflect.Type) bool { return t != nil && t.Kind() == reflect.Float32 }
+func isFloat64(t reflect.Type) bool { return t != nil && t.Kind() == reflect.Float64 }
 func isNumber(t reflect.Type) bool  { return isInt(t) || isFloat(t) || isComplex(t) }
-func isString(t reflect.Type) bool  { return t.Kind() == reflect.String }
+func isString(t reflect.Type) bool  { return t != nil && t.Kind() == reflect.String }

